ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs

```diff
@@ -2,6 +2,8 @@
 {-# LANGUAGE ScopedTypeVariables        #-}
 {-# LANGUAGE TypeFamilies               #-}
 {-# LANGUAGE BangPatterns               #-}
+{-# LANGUAGE AllowAmbiguousTypes        #-}
+{-# LANGUAGE TypeApplications           #-}
```

Comment:
Ugg. Do we have to?

Reply:
I suppose this is because `BlockPoint block` does not contain a `block`, only a `HeaderHash block`, and `HeaderHash` is not injective. Previously, `fetchDecisions` had a `(Point block -> Bool)` parameter, but now it's `(BlockPoint block -> Bool)`, which makes more sense, because obviously you have the origin point. Why did it work with the old `Point block` though? That one has a `ChainHash block` which also does not actually determine the `block` type, only its image under `HeaderHash`.

Reply:
Alternatively, I could use `Proxy`s to get rid of the ambiguity. Would you prefer that?

Reply:
>  Why did it work with the old Point block though? That one has a ChainHash block which also does not actually determine the block type, only its image under HeaderHash.

I'm guessing because wrapping the `HeaderHash block` in a `newtype` or `data` with the `block` as a parameter makes in unambiguous.